tests/evolution/test_genetic_algorithm_semantic_integration.py

```
@@ -0,0 +1,263 @@
+            generations=2,  # Minimum allowed by validation
+            use_semantic_operators=True,
+            semantic_operator_threshold=0.9,  # High threshold to avoid semantic ops
+            mutation_rate=0.2,  # Lower rate
+            crossover_rate=0.2  # Lower rate
+        )
+
+        ga = GeneticAlgorithm(llm_provider=mock_llm_provider)
+
+        # Mock fitness evaluator 
+        with patch.object(ga.fitness_evaluator, 'calculate_population_diversity', return_value=0.8):
+            with patch.object(ga.fitness_evaluator, 'evaluate_population') as mock_eval:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)